File: Java/app/src/main/java/dev/ogabek/java/adapter/DepartmentAdapter.java
package dev.ogabek.java.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import dev.ogabek.java.R;
import dev.ogabek.java.model.Department;

public class DepartmentAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {

    List<Department> departments;

    public DepartmentAdapter(List<Department> departments) {
        this.departments = departments;
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_department, parent, false);
        return new DepartmentViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        Department department = departments.get(position);

        if (holder instanceof DepartmentViewHolder) {
            ((DepartmentViewHolder) holder).photo.setImageResource(department.getPhoto());
            ((DepartmentViewHolder) holder).title.setText(department.getTitle());
        }
    }

    @Override
    public int getItemCount() {
        return departments.size();
    }

    private static class DepartmentViewHolder extends RecyclerView.ViewHolder {

        ImageView photo;
        TextView title;

        public DepartmentViewHolder(View view) {
            super(view);

            photo = view.findViewById(R.id.iv_department_photo);
            title = view.findViewById(R.id.tv_department_title);
        }
    }
}

File: Kotlin/app/src/main/java/dev/ogabek/kotlin/adapter/DepartmentAdapter.kt
package dev.ogabek.kotlin.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import dev.ogabek.kotlin.R
import dev.ogabek.kotlin.model.Department

class DepartmentAdapter(var departments: List<Department>) :
    RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view: View =
            LayoutInflater.from(parent.context).inflate(R.layout.item_department, parent, false)
        return DepartmentViewHolder(view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val department: Department = departments[position]
        if (holder is DepartmentViewHolder) {
            holder.photo.setImageResource(department.photo)
            holder.title.setText(department.title)
        }
    }

    override fun getItemCount(): Int {
        return departments.size
    }

    private class DepartmentViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var photo: ImageView = view.findViewById(R.id.iv_department_photo)
        var title: TextView = view.findViewById(R.id.tv_department_title)

    }

}